pkg/resourceutils/updatereplicas.go

```diff
@@ -0,0 +1,346 @@
+	updateCRDObjects(namespace, activate, printFunc, config)
+	updateCronJobObjects(namespace, activate, printFunc)
+}
+
+func updateStatefulSets(namespace string, activate bool, printFunc func(string, string)) {
+	statefulSets, err := apps.Instance().ListStatefulSets(namespace, metav1.ListOptions{})
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+	for _, statefulSet := range statefulSets.Items {
+		if replicas, update := getUpdatedReplicaCount(statefulSet.Annotations, activate, printFunc); update {
+			statefulSet.Spec.Replicas = &replicas
+			_, err := apps.Instance().UpdateStatefulSet(&statefulSet)
+			if err != nil {
+				printFunc(fmt.Sprintf("Error updating replicas for statefulset %v/%v : %v", statefulSet.Namespace, statefulSet.Name, err), "err")
+				continue
+			}
+			printFunc(fmt.Sprintf("Updated replicas for statefulset %v/%v to %v", statefulSet.Namespace, statefulSet.Name, replicas), "out")
+		}
+
+	}
+}
+
+func updateDeployments(namespace string, activate bool, printFunc func(string, string)) {
+	deployments, err := apps.Instance().ListDeployments(namespace, metav1.ListOptions{})
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+	for _, deployment := range deployments.Items {
+		if replicas, update := getUpdatedReplicaCount(deployment.Annotations, activate, printFunc); update {
+			deployment.Spec.Replicas = &replicas
+			_, err := apps.Instance().UpdateDeployment(&deployment)
+			if err != nil {
+				printFunc(fmt.Sprintf("Error updating replicas for deployment %v/%v : %v", deployment.Namespace, deployment.Name, err), "err")
+				continue
+			}
+			printFunc(fmt.Sprintf("Updated replicas for deployment %v/%v to %v", deployment.Namespace, deployment.Name, replicas), "out")
+		}
+	}
+}
+
+func updateDeploymentConfigs(namespace string, activate bool, printFunc func(string, string)) {
+	deployments, err := openshift.Instance().ListDeploymentConfigs(namespace)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			util.CheckErr(err)
+		}
+		return
+	}
+	for _, deployment := range deployments.Items {
+		if replicas, update := getUpdatedReplicaCount(deployment.Annotations, activate, printFunc); update {
+			deployment.Spec.Replicas = replicas
+			_, err := openshift.Instance().UpdateDeploymentConfig(&deployment)
+			if err != nil {
+				printFunc(fmt.Sprintf("Error updating replicas for deploymentconfig %v/%v : %v", deployment.Namespace, deployment.Name, err), "err")
+				continue
+			}
+			printFunc(fmt.Sprintf("Updated replicas for deploymentconfig %v/%v to %v", deployment.Namespace, deployment.Name, replicas), "out")
+		}
+	}
+}
+
+func updateCRDObjects(ns string, activate bool, printFunc func(string, string), config *rest.Config) {
+	crdList, err := storkops.Instance().ListApplicationRegistrations()
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+	configClient, err := k8sdynamic.NewForConfig(config)
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+	ruleset := inflect.NewDefaultRuleset()
```

Comment:
@diptiranjanpx Has a PR which moves this into a common function. Either one of you will have to rebase your changes

Reply:
I will fix this in the horntail-failover branch